internal/langserver/handlers/initialize.go

```diff
@@ -188,8 +105,150 @@ func (svc *service) Initialize(ctx context.Context, params lsp.InitializeParams)
+		}
+	}
+
+	// Walker runs asynchronously so we're intentionally *not*
+	// passing the request context here
+	walkerCtx := context.Background()
+	err = svc.walker.StartWalking(walkerCtx)
+	if err != nil {
+		return serverCaps, err
+	}
+
+	return serverCaps, nil
+}
+
+func setupTelemetry(expClientCaps lsp.ExpClientCapabilities, svc *service, ctx context.Context, properties map[string]interface{}) {
+	if tv, ok := expClientCaps.TelemetryVersion(); ok {
+		svc.logger.Printf("enabling telemetry (version: %d)", tv)
+		err := svc.setupTelemetry(tv, svc.server)
+		if err != nil {
+			svc.logger.Printf("failed to setup telemetry: %s", err)
+		}
+		svc.logger.Printf("telemetry enabled (version: %d)", tv)
+	}
+	defer svc.telemetry.SendEvent(ctx, "initialize", properties)
+}
+
+func getProperties(out *settings.DecodedOptions) map[string]interface{} {
+	properties := map[string]interface{}{
+		"experimentalCapabilities.referenceCountCodeLens": false,
+		"options.rootModulePaths":                         false,
+		"options.excludeModulePaths":                      false,
+		"options.commandPrefix":                           false,
+		"options.ignoreDirectoryNames":                    false,
+		"options.experimentalFeatures.validateOnSave":     false,
+		"options.terraformExecPath":                       false,
+		"options.terraformExecTimeout":                    "",
+		"options.terraformLogFilePath":                    false,
+		"root_uri":                                        "dir",
+		"lsVersion":                                       "",
+	}
+
+	properties["options.rootModulePaths"] = len(out.Options.ModulePaths) > 0
+	properties["options.excludeModulePaths"] = len(out.Options.ExcludeModulePaths) > 0
+	properties["options.commandPrefix"] = len(out.Options.CommandPrefix) > 0
+	properties["options.ignoreDirectoryNames"] = len(out.Options.IgnoreDirectoryNames) > 0
+	properties["options.experimentalFeatures.prefillRequiredFields"] = out.Options.ExperimentalFeatures.PrefillRequiredFields
+	properties["options.experimentalFeatures.validateOnSave"] = out.Options.ExperimentalFeatures.ValidateOnSave
+	properties["options.terraformExecPath"] = len(out.Options.TerraformExecPath) > 0
+	properties["options.terraformExecTimeout"] = out.Options.TerraformExecTimeout
+	properties["options.terraformLogFilePath"] = len(out.Options.TerraformLogFilePath) > 0
+	return properties
+}
+
+func resolvePath(rootDir, rawPath string) (string, error) {
+	path, err := homedir.Expand(rawPath)
+	if err != nil {
+		return "", err
+	}
+
+	if !filepath.IsAbs(path) {
+		path = filepath.Join(rootDir, rawPath)
+	}
+
+	return cleanupPath(path)
+}
+
+func cleanupPath(path string) (string, error) {
+	absPath, err := filepath.Abs(path)
+	return toLowerVolumePath(absPath), err
+}
+
+func toLowerVolumePath(path string) string {
+	volume := filepath.VolumeName(path)
+	return strings.ToLower(volume) + path[len(volume):]
+}
+
+func buildInitializeResult(ctx context.Context) lsp.InitializeResult {
```

Comment:
```suggestion
func initializeResult(ctx context.Context) lsp.InitializeResult {
```
It's not particularly _strong_ convention, but AFAIK it is a convention to prefix function names with the verbs like this in Go. e.g. getters and setters typically aren't prefixed with `get` or `set` in Go, unlike in other languages.

Reply:
Applied edit